models/ActivityModel.ts

```diff
@@ -11,7 +11,7 @@ export class ActivityModel extends BaseEntity {
 
   @ManyToOne((type) => UserModel, (user) => user.activities, { nullable: false })
   @JoinColumn({ name: 'user' })
-  @Index('public_activities_by_user_index', { where: 'public IS true' })
+  @Index('public_activities_by_user_index', { where: 'scope = \'PUBLIC\'' })
   user: UserModel;
```

Comment:
Should this partial index be `PUBLIC or PRIVATE` or just `PUBLIC`? A user (after redesign) can only view other users' public activities, but can view their own public and private activities, so do we need to optimize for the latter?

Reply:
Maybe index on both `user` and `scope`?

Reply:
WHERE can be for all non-hidden activities.

Reply:
Something like `visible_activities_by_user_index` would probably be a better index name if the WHERE's for non-hidden activities.